### PR TITLE
fix tab issues in actionbar

### LINKED ISSUE #219199: Tab gets stuck in Problems filter input
> The code I linked is related to a separate issue that I saw, which probably isn't really related to what you fixed. But if I put focus in the Problems or Debug Console filter input, and do some combination of tab out of the input, shift+tab back into it, type in it, then at some point focus gets stuck in the widget and 'tab' doesn't move focus out to the next toolbar. 

 _Originally posted in [#211979](https://github.com/microsoft/vscode/issues/211979#issuecomment-2187963936)_

### PR DESCRIPTION
**Pull request by meganrogge**
fixes #219199 

https://github.com/user-attachments/assets/6359e0d0-c714-4e8c-8679-9ae400bdddf6

### CODE REVIEW

**Review by mjbvz — DISMISSED** [review]

**Comment by roblourens** [comment]
I can help test but I really don't know anything about this code

**Comment by bpasero** [comment]
This predates my memory, would it help to blame it to its origin, possibly from our older DevOps repo?

**Comment by meganrogge** [comment]
I found via gitlens that @connor4312 introduced the `forceFocus` concept, so he would be a good reviewer. Looking at his PR now

https://github.com/microsoft/vscode/pull/136398

**Comment by meganrogge** [comment]
It seems re-adding `forceFocus` is fine as long as that's passed in as `true` for the `focusNext` call

**Comment by meganrogge** [comment]
I notice that since your PR, there's been a change where this used to be getting set to `undefined`, but is now set to `this.focusedItem`

https://github.com/microsoft/vscode/blob/0bdf4c00861d440943b6e391a250000d066c0b4d/src/vs/base/browser/ui/actionbar/actionbar.ts#L212

**Comment by meganrogge** [comment]
I think the issue is here:

https://github.com/microsoft/vscode/blob/0bdf4c00861d440943b6e391a250000d066c0b4d/src/vs/base/browser/ui/actionbar/actionbar.ts#L285

**Comment by meganrogge** [comment]
Screenshot 2024-08-01 at 12 27 01 PM

**Comment by meganrogge** [comment]
the update of the focused item's index is occurring before the actually focusing has taken place, which then prevents the focus from occurring

Screenshot 2024-08-01 at 12 47 16 PM

File: src/vs/base/browser/ui/menu/menu.ts [hunk 1]
```diff
@@ -376,7 +376,7 @@ export class Menu extends ActionBar {
 	}
 
 	protected override updateFocus(fromRight?: boolean): void {
-		super.updateFocus(fromRight, true, true);
+		super.updateFocus(fromRight, true);
```
Comment (meganrogge) [inline src/vs/base/browser/ui/menu/menu.ts:379]:
```suggestion
		super.updateFocus(fromRight, true, true);
```

File: src/vs/base/browser/ui/actionbar/actionbar.ts [hunk 2]
```diff
@@ -563,6 +567,7 @@ export class ActionBar extends Disposable implements IActionRunner {
 			if (!focusItem) {
 				this.actionsList.focus({ preventScroll });
 				this.previouslyFocusedItem = undefined;
+				return;
```
Comment (meganrogge) [inline src/vs/base/browser/ui/actionbar/actionbar.ts:570]:
```suggestion
```

**Review by connor4312 — APPROVED** [review]

**Pull request merged**